### PR TITLE
server: do not log on handshake error

### PR DESCRIPTION
**Pull request by coocood**
Load balancers connects to TiDB server and close immediately, we don't need to log them.

### CODE REVIEW

**Comment by coocood** [comment]
verified by connecting to TiDB server and close with telnet, no logs anymore.
@shenli @zz-jason @zimulala PTAL

**Comment by ngaut** [comment]
LGTM

File: server/server.go [hunk 1]
```diff
@@ -293,19 +292,18 @@ func (s *Server) Close() {
...
-	defer func() {
-		log.Infof("[%d] close connection", conn.connectionID)
-	}()
-
 	if err := conn.handshake(); err != nil {
 		// Some keep alive services will send request to TiDB and disconnect immediately.
-		// So we use info log level.
-		log.Infof("handshake error %s", errors.ErrorStack(err))
+		// So we only record metrics.
+		metrics.HandShakeErrorCounter.Inc()
```
Comment (shenli) [inline server/server.go]: Remove line 300?

> terror.Log(errors.Trace(err))

File: terror/terror.go [hunk 2]
```diff
@@ -278,7 +278,7 @@ func (e *Error) getMySQLErrorCode() uint16 {
 	}
 	code, ok := codeMap[e.code]
 	if !ok {
-		log.Warnf("Unknown error class: %v code: %v", e.class, e.code)
+		log.Debugf("Unknown error class: %v code: %v", e.class, e.code)
```
Comment (shenli) [inline terror/terror.go]: Why change this?

File: server/server.go (hunk 1 above)
Comment (coocood) [reply server/server.go]: Then errcheck would fail.
`Close` normally doesn't return any error.

**Pull request merged**